src/main/java/io/rocketpartners/cloud/action/sql/SqlDb.java

```diff
@@ -186,7 +194,11 @@ protected void shutdown0()
 
       String sql = (String) Chain.peek().remove("select");
       if (Utils.empty(sql))
+      {
+         if (table == null)
+            throw new ApiException(SC.SC_400_BAD_REQUEST, "Table missing");
          sql = " SELECT * FROM " + quoteCol(table.getName());
+      }
```

Comment:
if we didn't have a select statement from the chain, we were selecting from the table (provided from the collection). If table is null, seems like we don't have a collection and we're in undefined territory.